08-transport.md

```diff
@@ -402,6 +407,43 @@ construction, and 16 bytes for a final authenticating tag.
 10. `rn = 0, sn = 0`
      * The sending and receiving nonces are initialized to 0.
 
+## Alternate Transport Layers: WebSocket
+
+Normally the transport protocol defined here is performed over TCP/IP,
+but it can also be performed over other underlying transports, such as
+the WebSocket protocol as specified in
+RFC6455<sup>[4](#reference-4)</sup>.
+
+A client may connect to a node and initiate a WebSocket; this will
+normally fail as the WebSocket protocol begins with a "GET" request,
+which is trivially distinguishable from a valid handshake.  However,
+the node may also allow it and operate the protocol over binary
+WebSocket frames.  The `option_websocket` feature allows nodes to
+advertise this, but not all nodes send node_announcements, so it is
+not required before attempting a WebSocket connection.
+
+
+### Requirements
+
+The initiator:
+- MAY attempt to initiate an unencrypted WebSocket as specified in RFC6455<sup>[4](#reference-4)</sup>:
+  - MUST send at least 50 bytes before awaiting a response.
+  - MUST abort the connection attempt if WebSocket upgrade fails.
+  - MUST begin the [Handshake Exchange](#handshake-exchange) as initiator 
+    as soon as upgrade succeeds.
+
+The responder:
+- if it supports `option_websocket`:
+  - SHOULD set `option_websocket` in its node announcements
+  - MUST attempt WebSocket upgrade if the Act 1 handshake it receives
```

Comment:
See comment above re separation of the layers. When nodes connect over Tor today, `brontide` implementations don't need to do anything different, and we can inherit that behavior here by defining the web sockets feature as a new _address_ type.

Reply:
Also by having it as a new address type, nodes can advertise another addr here, which might actually just be a bridge they're using.